addons/crm_iap_lead_enrich/models/crm_lead.py

```diff
@@ -145,3 +146,29 @@ def _iap_enrich_from_response(self, iap_response):
                 values=template_values,
                 subtype_id=self.env.ref('mail.mt_note').id
             )
+
+    def write(self, values):
+        self.ensure_one()
+        if self.iap_enrich_done:
+            dbuuid = self.env['ir.config_parameter'].sudo().get_param('database.uuid')
+            endpoint = self.env['ir.config_parameter'].sudo().get_param('enrich.endpoint', self._DEFAULT_IAP_ENDPOINT) + '/iap/directory/1/vote'
+            if self.website:
+                company_domain = tools.url_domain_extract(self.website)
+            elif self.email:
+                company_domain = tools.email_domain_extract(self.email)
+            else:
+                return super(Lead, self).write(values)
+
+            for key in values.keys():
+                params = {
+                    'db_uuid': dbuuid,
+                    'domain': company_domain,
+                    'field': key,
+                    'value': values[key],
+                }
+                try:
+                    iap_tools.iap_jsonrpc(endpoint, params=params, timeout=1)
```

Comment:
@fda-odoo What do you think about the 1s timeout ?

Reply:
1 second seems ok for me, but why do you do a call per field? Can't you group them in one call? Or it will be 1 second per field!